encode.go

```diff
@@ -367,8 +367,14 @@ func ParseTimestamp(currentLocation *time.Location, str string) (time.Time, erro
 	timeSep := daySep + 3
 	day := p.mustAtoi(str, daySep+1, timeSep)
 
+	isBC := len(str) > 3 && str[len(str)-3:] == " BC"
```

Comment:
isn't this `strings.HasSuffix(str, " BC")`? perhaps write:
```golang
minLen := monSep+len("01-01")+1

const suffixBC = " BC"
isBC := strings.HasSuffix(str, suffixBC)
if isBC {
  minLen += len(suffixBC)
}
var ... 
if len(str) > minLen {
...
```

Reply:
great recommendation, I have update the PR to follow this approach. I opted not to use the const to follow the pattern in the surrounding code.